### PR TITLE
[HIU_API_AUTH_BUG] | Mahendra/Gayatri | Fix authentication bug on HIP APIs

### PR DESCRIPTION
**Pull request by tw-mahendra**

### CODE REVIEW

File: src/main/java/in/org/projecteka/hiu/HiuConfiguration.java [hunk 1]
```diff
@@ -434,21 +433,9 @@ public CentralRegistryTokenVerifier centralRegistryTokenVerifier(@Qualifier("cen
     }
 
     @Bean
-    @ConditionalOnProperty(value = "hiu.loginMethod", havingValue = "jwt")
+    @ConditionalOnProperty(value = "hiu.loginMethod", havingValue = "jwt", matchIfMissing = true)
```
Comment (ganesanarun) [inline src/main/java/in/org/projecteka/hiu/HiuConfiguration.java]: you can remove this completely

File: src/main/java/in/org/projecteka/hiu/HiuConfiguration.java [hunk 2]
```diff
@@ -434,21 +433,9 @@ public CentralRegistryTokenVerifier centralRegistryTokenVerifier(@Qualifier("cen
...
-    @ConditionalOnProperty(value = "hiu.loginMethod", havingValue = "base64")
-    public Authenticator base64Authenticator() {
-        return new Base64Authenticator();
-    }
-
-    @Bean
-    @ConditionalOnProperty(value = "hiu.loginMethod", havingValue = "both", matchIfMissing = true)
-    public Authenticator userBase64Authenticator(byte[] sharedSecret) throws JOSEException {
-        return new UserBase64Authenticator(new Base64Authenticator(), new UserAuthenticator(sharedSecret));
+        return new UserAuthenticator(new JwtAuthenticator(sharedSecret));
```
Comment (ganesanarun) [inline src/main/java/in/org/projecteka/hiu/HiuConfiguration.java]: You don't need this wrapper anymore.

File: src/main/java/in/org/projecteka/hiu/HiuConfiguration.java (hunk 1 above)
Comment (gayatrijagtap) [reply src/main/java/in/org/projecteka/hiu/HiuConfiguration.java]: removed

File: src/main/java/in/org/projecteka/hiu/HiuConfiguration.java (hunk 2 above)
Comment (gayatrijagtap) [reply src/main/java/in/org/projecteka/hiu/HiuConfiguration.java]: removed the wrapper

**Review by ganesanarun — APPROVED** [review]

**Pull request merged**